### PR TITLE
[10.0] [MIG] stock_picking_variant_mgmt: Migration to 10.0

### PR DESCRIPTION
**Pull request by pedrobaeza**
* Standard migration
* Tests adapted

@Tecnativa

### CODE REVIEW

**Review by carlosdauden — COMMENTED** [review]
Code LGTM.

Testing in runbot shows intermittent error when click in new matrix button of picking line:
Uncaught TypeError: Cannot read property 'type' of undefined
`http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/474-bfd16b8/web.assets_backend.js:4101
Traceback:
TypeError: Cannot read property 'type' of undefined
    at Class.format_xy_value (http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/474-bfd16b8/web.assets_backend.js:4101:72)
    at Class.compute_totals (http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/474-bfd16b8/web.assets_backend.js:4105:142)
    at Class.start (http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/474-bfd16b8/web.assets_backend.js:4112:90)
    at Class.prototype.(anonymous function) [as start] (http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/468-bf677a6/web.assets_common.js:3010:559)
    at http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/468-bf677a6/web.assets_common.js:3113:963
    at http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/468-bf677a6/web.assets_common.js:547:681
    at fire (http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/468-bf677a6/web.assets_common.js:541:299)
    at Object.add [as done] (http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/468-bf677a6/web.assets_common.js:542:467)
    at Array.<anonymous> (http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/468-bf677a6/web.assets_common.js:547:649)
    at Function.each (http://3308592-84-152fd1.runbot1.odoo-community.org/web/content/468-bf677a6/web.assets_common.js:370:767)`

Can be related with known issues, but v10?

**Comment by pedrobaeza** [comment]
That's because it's not OCB. Please read known issues section of the README.

**Comment by carlosdauden** [comment]
Precisely because I read known issues I asked this:

> Can be related with known issues, but v10?

This section has 1 v8 PR reference and 1 v9 PR reference:
 - https://github.com/odoo/odoo/pull/13557
 - https://github.com/odoo/odoo/pull/13635

**Comment by pedrobaeza** [comment]
For v10 I declined to do it again, as it was rejected in previous versions.

**Review by carlosdauden — APPROVED** [review]

File: stock_picking_variant_mgmt/README.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+* The inline button for modifying quantities for an existing line won't
+  work correctly until these 2 PRs are merged in Odoo:
+
+  * https://github.com/odoo/odoo/pull/13557
+  * https://github.com/odoo/odoo/pull/13635
+
+  The patches are already integrated on OCB.
+
+* Make this work with products with more than 1 attribute.
```
Comment (sergio-teruel) [inline stock_picking_variant_mgmt/README.rst]: Remove this, You can use with one attribute too.

File: stock_picking_variant_mgmt/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    'author': 'Tecnativa,'
+              'Odoo Community Association (OCA)',
+    'category': 'Inventory, Logistics, Warehousing',
+    'license': 'AGPL-3',
+    'website': 'https://www.tecnativa.com',
+    'depends': [
+        'stock',
+        'web_widget_x2many_2d_matrix',
+    ],
+    'demo': [],
```
Comment (sergio-teruel) [inline stock_picking_variant_mgmt/__openerp__.py]: Remove empty key

File: stock_picking_variant_mgmt/wizard/stock_manage_variant_view.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        <field name="model">stock.manage.variant</field>
+        <field name="arch" type="xml">
+            <form string="Manage Product Variants">
+                <group>
+                    <field name="product_tmpl_id"
+                           domain="[('attribute_line_ids', '!=', False)]"
+                           options="{'no_create': True}"
+                    />
+                </group>
+                <field name="variant_line_ids"
```
Comment (sergio-teruel) [inline stock_picking_variant_mgmt/wizard/stock_manage_variant_view.xml]: Put this field inside a group with nolabel, now the field appears in small area.

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: stock_picking_variant_mgmt/README.rst (hunk 1 above)
Comment (pedrobaeza) [reply stock_picking_variant_mgmt/README.rst]: No, the issue is correct. See that it puts "more than 1 attribute", which is the problem indeed in the transfer dialog. I will be any way more specific.

**Comment by pedrobaeza** [comment]
@sergio-teruel changes done

**Comment by sergio-teruel** [comment]
Only one question about JS error ... is not very usable.. ¿the v9 commit would be for v10?

**Comment by pedrobaeza** [comment]
In OCB v10 it's already integrated.

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**